homeassistant/components/rpi_gpio_pwm/number.py

```diff
@@ -0,0 +1,174 @@
+        elif driver_type == CONF_DRIVER_PCA9685:
+            if CONF_ADDRESS in number_conf:
+                opt_args["address"] = number_conf[CONF_ADDRESS]
+            driver = Pca9685Driver([pin], **opt_args)
+        else:
+            _LOGGER.error("Invalid driver type")
+            return
+
+        number = PwmNumber(hass, number_conf, driver)
+        numbers.append(number)
+
+    add_entities(numbers)
+
+
+class PwmNumber(NumberEntity, RestoreEntity):
+    """Representation of a simple  PWM output."""
+
+    def __init__(self, hass, config, driver):
+        """Initialize one-color PWM LED."""
+        self._driver = driver
+        self._config = config
+        self._hass = hass
+        self._attr_min_value = config[CONF_MINIMUM]
+        self._attr_max_value = config[CONF_MAXIMUM]
+        self._attr_step = config[CONF_STEP]
+        self._attr_mode = config[CONF_MODE]
+        self._attr_value = config[CONF_MINIMUM]
+
+    async def async_added_to_hass(self):
+        """Handle entity about to be added to hass event."""
+        await super().async_added_to_hass()
+        if last_state := await self.async_get_last_state():
+            await self.async_set_value(float(last_state.state))
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the number."""
+        return self._config[CONF_NAME]
+
+    @property
+    def frequency(self):
+        """Return PWM frequency."""
+        return self._config[CONF_FREQUENCY]
+
+    @property
+    def invert(self):
+        """Return if output is inverted."""
+        return self._config[CONF_INVERT]
+
+    @property
+    def capability_attributes(self) -> dict[str, Any]:
+        """Return capability attributes."""
+        attr = super().capability_attributes
+        attr[ATTR_FREQUENCY] = self.frequency
+        attr[ATTR_INVERT] = self.invert
+        return attr
+
+    async def async_set_value(self, value: float) -> None:
+        """Set new value."""
+        # Clip value to limits (don't know if this is required?)
+        if value < self._config[CONF_MINIMUM]:
+            value = self._config[CONF_MINIMUM]
+        if value > self._config[CONF_MAXIMUM]:
+            value = self._config[CONF_MAXIMUM]
+
+        # Scale range from 0..100 to 0..255 (gpio) or 0..65535 (pca9685)
+        max_pwm = 255.0
+        if self._config[CONF_DRIVER] == CONF_DRIVER_PCA9685:
+            max_pwm = 65535.0
+        # In case the invert bit is on, invert the value
+        used_value = value
+        if self._config[CONF_INVERT]:
+            used_value = 100.0 - value
+        # Scale to range of the driver
+        scaled_value = int(round((used_value / 100.0) * max_pwm))
+        # Set value to driver
+        self._driver._set_pwm([scaled_value])  # pylint: disable=W0212
+        self._attr_value = value
+        self.schedule_update_ha_state()
```

Comment:
I think we can write the state update immediately?
```suggestion
        self.async_write_ha_state()
```

Reply:
light.py contains also update_ha_state, shouln't we use write_ha_state there as well?

Reply:
Yeah, that's probably fine.
Please make sure to test though.

Reply:
Done in commit cecf42dc8f7477c7490f004b1e399c396a15b2ed